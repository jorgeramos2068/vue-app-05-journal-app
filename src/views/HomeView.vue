<template>
  <div class="home">
    <img alt="Vue logo" src="../assets/logo.png" />
    <h1>Hello World</h1>
    <button class="btn btn-primary" @click="goToDaybook()">Primary</button>
  </div>
</template>

<script>
export default {
  methods: {
    goToDaybook() {
      this.$router.push({ name: 'daybook' });
    },
  },
};
</script>
