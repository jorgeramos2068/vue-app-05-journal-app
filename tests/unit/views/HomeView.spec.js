import { shallowMount } from '@vue/test-utils';
import HomeView from '@/views/HomeView.vue';

describe('HomeView component', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallowMount(HomeView);
  });

  it('should match the snapshot', () => {
    expect(wrapper.html()).toMatchSnapshot();
  });

  it('should redirect to no-entry when a button is clicked', () => {
    const mockRouter = {
      push: jest.fn(),
    };
    const wrapper = shallowMount(HomeView, {
      global: { mocks: { $router: mockRouter } },
    });
    wrapper.find('button').trigger('click');
    expect(mockRouter.push).toHaveBeenCalledWith({ name: 'no-entry' });
  });
});
