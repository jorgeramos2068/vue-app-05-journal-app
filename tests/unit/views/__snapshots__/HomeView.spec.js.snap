// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HomeView component should match the snapshot 1`] = `
<div class="home"><img alt="Vue logo" src="">
  <h1>Hello World</h1><button class="btn btn-primary">Primary</button>
</div>
`;
